src/tests/test_aead.cpp

```diff
@@ -61,12 +67,20 @@ class AEAD_Tests final : public Text_Based_Test {
          result.test_eq("key is set", enc->has_keying_material(), true);
 
          if(is_siv == false) {
-            result.test_throws("Cannot process data until nonce is set (enc)", [&]() { enc->update(garbage); });
-            result.test_throws("Cannot process data until nonce is set (enc)", [&]() { enc->finish(garbage); });
+            result.test_throws("Cannot process data until nonce is set (enc)", [&]() {
+               auto garbage = get_garbage();
+               enc->update(garbage);
+            });
+            result.test_throws("Cannot process data until nonce is set (enc)", [&]() {
+               auto garbage = get_garbage();
+               enc->finish(garbage);
+            });
          }
 
          enc->set_associated_data(mutate_vec(ad, rng));
          enc->start(mutate_vec(nonce, rng));
+
+         auto garbage = get_garbage();
          enc->update(garbage);
```

Comment:
For OCB with a tag-length of 12 and an update_granularity of 16, the above-mentioned side effect (introduced in #4880) would result in a `garbage` vector that doesn't fit the update granularity and make this call to update fail unexpectedly.